contracts/protocol/libraries/logic/GenericLogic.sol

```diff
@@ -194,52 +195,32 @@ library GenericLogic {
                     params.userConfig.isUsingAsCollateral(vars.i)
                 ) {
-                    if (INToken(vars.xTokenAddress).getAtomicPricingConfig()) {
-                        if (vars.dynamicConfigs) {
-                            (
-                                vars.userBalanceInBaseCurrency,
-                                vars.dynamicLTV,
-                                vars.dynamicLiquidationThreshold
-                            ) = _getUserBalanceAndDynamicConfigsForAtomicERC721(
-                                params.user,
-                                currentReserve,
-                                params.oracle,
-                                vars.currentReserveAddress,
-                                vars.assetUnit
-                            );
-
-                            vars.liquidationThreshold = vars
-                                .dynamicLiquidationThreshold;
-                            vars.avgLtv += vars.dynamicLTV;
-                        } else {
-                            vars
-                                .userBalanceInBaseCurrency = _getUserBalanceForAtomicERC721(
-                                params.user,
-                                vars.xTokenAddress,
-                                params.oracle,
-                                vars.currentReserveAddress,
-                                vars.assetUnit
-                            );
-                        }
-                    } else {
-                        vars.assetPrice = _getAssetPrice(
-                            params.oracle,
-                            vars.currentReserveAddress
+                    vars.isAutomicPrice = INToken(vars.xTokenAddress)
+                        .getAtomicPricingConfig();
+                    if (vars.isAutomicPrice && vars.dynamicConfigs) {
+                        (
+                            vars.userBalanceInBaseCurrency,
+                            vars.dynamicLTV,
+                            vars.dynamicLiquidationThreshold
+                        ) = _getUserBalanceAndDynamicConfigsForAtomicERC721(
+                            params,
+                            vars,
+                            currentReserve.dynamicConfigsStrategyAddress
                         );
 
+                        vars.liquidationThreshold = vars
+                            .dynamicLiquidationThreshold;
+                        vars.avgLtv += vars.dynamicLTV;
+                    } else {
                         vars
                             .userBalanceInBaseCurrency = _getUserBalanceForERC721(
-                            params.user,
-                            vars.xTokenAddress,
-                            vars.assetUnit,
-                            vars.assetPrice
+                            params,
+                            vars
                         );
-                    }
 
-                    if (!vars.dynamicConfigs) {
```

Comment:
removing this will cause a problem since the dynamicLiquidationThreshold is technically the price of each token multiplied by the its liquidation threshold. did you run the tests after this commit?

Reply:
I moved the check for `vars.dynamicConfigs` to the else branch. and yeah unit test is ok.